File: Algorithms/1370.上升下降字符串/sortString.java
class Solution {
    public String sortString(String s) {
        int[] count = new int[26];
        for (int i = 0; i < s.length(); ++i)
            ++count[s.charAt(i) - 'a'];

        StringBuilder res = new StringBuilder();
        while (res.length() < s.length()) {
            for (int i = 0; i <= 25; ++i)
                if (count[i] > 0) {
                    res.append((char) ('a' + i));
                    --count[i];
                }
            for (int i = 25; i >= 0; --i)
                if (count[i] > 0) {
                    res.append((char) ('a' + i));
                    --count[i];
                }
        }
        return res.toString();
    }
}

File: Algorithms/1370/sortString.py
from collections import Counter

class Solution:
    def sortString(self, s: str) -> str:
        count = Counter(s)
        sorted_letters = list(sorted(count))
        res = []
        n = len(s)
        while n:
            for k in sorted_letters:
                if count[k]:
                    res.append(k)
                    count[k] -= 1
                    n -= 1
            for k in reversed(sorted_letters):
                if count[k]:
                    res.append(k)
                    count[k] -= 1
                    n -= 1
        return ''.join(res)
